File: strategy/src/main/java/com/nearsoft/workshop/strategy/character/Troll.java
package com.nearsoft.workshop.strategy.character;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class Troll extends Character {
    private final Logger log = LoggerFactory.getLogger(this.getClass());

    @Override
    public void fight() {
        log.debug(String.format("Trololol, I'm %s", weaponBehaviour.useWeapon()));
    }
}

File: strategy/src/main/scala/com/nearsoft/workshop/strategy/scala/character/Troll.scala
package com.nearsoft.workshop.strategy.scala.character

import org.slf4j.LoggerFactory

class Troll extends Character {

  val log = LoggerFactory.getLogger(classOf[King])

  override def fight = {
    log.debug(s"Trololol, I'm ${useWeapon}")
  }
}
